PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h

```diff
@@ -2699,164 +2961,455 @@ void ResetEventByEventQuantities()
+          return kFALSE;
+        }
       }
-      return kFALSE;
     }
+
     //   *) Vertex_z:
-    if (ec.fUseEventCuts[eVertex_z] && (collision.posZ() < ec.fdEventCuts[eVertex_z][eMin] || collision.posZ() > ec.fdEventCuts[eVertex_z][eMax])) {
-      if (tc.fVerbose) {
-        LOGF(info, "\033[1;31m%s eVertex_z\033[0m", __FUNCTION__);
+    if (ec.fUseEventCuts[eVertex_z]) {
+      if (cutModus == eCutCounterBinning) {
+        EventCut(eRec, eVertex_z, eCutCounterBinning);
+      } else if (collision.posZ() < ec.fdEventCuts[eVertex_z][eMin] || collision.posZ() > ec.fdEventCuts[eVertex_z][eMax]) {
+        if (!EventCut(eRec, eVertex_z, cutModus)) {
+          return kFALSE;
+        }
       }
-      return kFALSE;
     }
+
     //   *) NContributors:
-    if (ec.fUseEventCuts[eNContributors] && (collision.numContrib() < ec.fdEventCuts[eNContributors][eMin] || collision.numContrib() > ec.fdEventCuts[eNContributors][eMax])) {
-      if (tc.fVerbose) {
-        LOGF(info, "\033[1;31m%s eNContributors\033[0m", __FUNCTION__);
+    if (ec.fUseEventCuts[eNContributors]) {
+      if (cutModus == eCutCounterBinning) {
+        EventCut(eRec, eNContributors, eCutCounterBinning);
+      } else if (collision.numContrib() < ec.fdEventCuts[eNContributors][eMin] || collision.numContrib() > ec.fdEventCuts[eNContributors][eMax]) {
+        if (!EventCut(eRec, eNContributors, cutModus)) {
+          return kFALSE;
+        }
       }
-      return kFALSE;
     }
-    // TBI 20231106 continue here with other event cuts on reconstructed info
 
-    // ... and corresponding MC truth simulated ( see https://github.com/AliceO2Group/O2Physics/blob/master/Tutorials/src/mcHistograms.cxx ):
+    //   *) SelectedEvents: => this event cut is implemented directly in Steer(...)
+
+    // ...
+
+    // ... and corresponding MC truth simulated:
+    // See https://github.com/AliceO2Group/O2Physics/blob/master/Tutorials/src/mcHistograms.cxx
+    // See https://aliceo2group.github.io/analysis-framework/docs/datamodel/ao2dTables.html#montecarlo
     if constexpr (rs == eRecAndSim || rs == eRecAndSim_Run2 || rs == eRecAndSim_Run1) {
       if (!collision.has_mcCollision()) {
         LOGF(warning, "No MC collision for this collision, skip..."); // TBI 20231106 re-think. I shouldn't probably get to this point, if MC truth info doesn't exist for this collision
         return kFALSE;
       }
 
-      // TBI 20231106 here I cat cut directly on corresponding MC truth simulated, e.g. on collision.mcCollision().posZ(), if necessary
+      // In this branch I can cut additionally and directly on corresponding MC truth simulated, e.g. on collision.mcCollision().posZ().
+      // In case I implement something here, remember to switch from eRec to eSim when calling e.g. EventCut(...)
+      cout << collision.mcCollision().posZ() << endl;            // TBI 20240509 remove
+      cout << collision.mcCollision().impactParameter() << endl; // TBI 20240509 remove
```

Comment:
This has to be replaced in the next iteration

Reply:
Thanks, noted! I overlooked this one, but it's removed now! I will commit this change alongside the next PR within the next few days or so.